app/assets/src/components/views/SampleViewV2/ReportTableV2.jsx

```diff
@@ -0,0 +1,404 @@
+        cellDataGetter: ({ rowData }) => {
+          return [
+            (rowData.nt || {}).percent_identity || 0,
+            (rowData.nr || {}).percent_identity || 0,
+          ];
+        },
+        cellRenderer: ({ cellData }) =>
+          this.renderNtNrDecimalValues({ cellData, decimalPlaces: 1 }),
+        dataKey: "percent_identity",
+        label: "%id",
+        sortFunction: ({ data, sortDirection }) =>
+          this.nestedNtNrSortFunction({
+            data,
+            sortDirection,
+            path: ["percent_identity"],
+            nullValue: 0,
+            limits: NUMBER_NULL_VALUES,
+          }),
+        width: 60,
+      },
+      {
+        cellDataGetter: ({ rowData }) => {
+          return [
+            (rowData.nt || {}).alignment_length || 0,
+            (rowData.nr || {}).alignment_length || 0,
+          ];
+        },
+        cellRenderer: ({ cellData }) =>
+          this.renderNtNrDecimalValues({ cellData, decimalPlaces: 1 }),
+        dataKey: "alignment_length",
+        label: "L",
+        sortFunction: ({ data, sortDirection }) =>
+          this.nestedNtNrSortFunction({
+            data,
+            sortDirection,
+            path: ["alignment_length"],
+            nullValue: 0,
+            limits: NUMBER_NULL_VALUES,
+          }),
+        width: 60,
+      },
+      {
+        cellDataGetter: ({ rowData }) => {
+          return [
+            (rowData.nt || {}).e_value || 0,
+            (rowData.nr || {}).e_value || 0,
+          ];
+        },
+        cellRenderer: ({ cellData }) =>
+          this.renderNtNrDecimalValues({ cellData, decimalPlaces: 1 }),
+        dataKey: "e_value",
+        label: "log(1/E)",
+        sortFunction: ({ data, sortDirection }) =>
+          this.nestedNtNrSortFunction({
+            data,
+            sortDirection,
+            path: ["e_value"],
+            nullValue: 0,
+            limits: NUMBER_NULL_VALUES,
+          }),
+        width: 60,
+      },
+      {
+        dataKey: "ntnrSelector",
+        disableSort: true,
+        headerClassName: cs.ntnrSelectorHeader,
+        headerRenderer: this.renderNtNrSelector,
+        width: 40,
+      },
+    ];
+  }
+
+  nestedSortFunction = ({ data, path, sortDirection, nullValue, limits }) => {
+    // Uses lodash's orderBy function.
+    // It uses a sorting key that enables nested sorting of genus and species, while guaranteeing that
+    // genus is always on top of its children species
+    return orderBy(
+      [
+        rowData =>
+          rowData.genus
+            ? getOr(nullValue, ["genus"].concat(path), rowData)
+            : getOr(nullValue, path, rowData),
+        // this value guarantees that we keep species within their genus, even if the first value is duplicated
+        rowData => (rowData.genus ? rowData.genus.taxId : rowData.taxId),
+        rowData =>
+          rowData.genus
+            ? getOr(nullValue, path, rowData)
+            : sortDirection === "asc"
```

Comment:
I'm a little confused why you need this `limits[0] : limits[1]` here. If `rowData.genus` doesn't exist, then it's a genus row, so it should already be sorted by the first key, right? It seems like it would be okay to have it always be the same number since it doesn't matter, instead of two numbers (limit[0] and limit[1]).

Reply:
Maybe my confusion can suggest an appropriate comment to put here.

Reply:
It matters because it is nested. No matter asc or desc you want the genus on top of their species.